crates/ide_db/src/helpers.rs

```diff
@@ -161,9 +161,7 @@ pub fn for_each_tail_expr(expr: &ast::Expr, cb: &mut dyn FnMut(&ast::Expr)) {
                 ) => return cb(expr),
 
                 Some(ast::BlockModifier::Label(label)) => {
-                    for_each_break_expr(Some(label), b.stmt_list(), &mut |b| {
-                        cb(&ast::Expr::BreakExpr(b))
-                    });
+                    for_each_break_and_continue_expr(Some(label), b.stmt_list(), &mut |b| cb(&b));
```

Comment:
This is incorrect, `continue` is not to be considered a tail expression as it doesn't break off a loop